### PR TITLE
reserving-syntax.md: Expand and add detail

### PR DESCRIPTION
**Pull request by joshtriplett**
cc https://github.com/rust-lang/edition-guide/issues/242

### CODE REVIEW

File: src/rust-2021/reserving-syntax.md [hunk 1]
```diff
@@ -2,44 +2,47 @@
 
 ## Summary
 
-- `any_prefix#..`, `any_prefix".."`, and `any_prefix'..'` are reserved syntax, and no longer tokenize.
+- `any_prefix#`, `any_prefix"..."`, and `any_prefix'...'` are now reserved
```
Comment (rylev) [inline src/rust-2021/reserving-syntax.md:5]: I'm not sure if the use of `any_syntax` is clear. Perhaps we describe it in words (e.g., "the use of any syntax followed by a "#" or literals enclosed in single or double quotes (e.g., `foo#`, `bar'baz'` , `qux"corge"`) is now reserved")

**Review by rylev — APPROVED** [review]
Noted a small concern, but nothing blocking merging!

File: src/rust-2021/reserving-syntax.md (hunk 1 above)
Comment (nikomatsakis) [reply src/rust-2021/reserving-syntax.md:5]: I agree that we should at least define `any_prefix`

File: src/rust-2021/reserving-syntax.md [hunk 2]
```diff
@@ -2,44 +2,47 @@
...
+  syntax, and no longer tokenize.
 - This is mostly relevant to macros. E.g. `quote!{ #a#b }` is no longer accepted.
 - It doesn't treat keywords specially, so e.g. `match".." {}` is no longer accepted.
 - Insert whitespace to avoid errors.
```
Comment (nikomatsakis) [inline src/rust-2021/reserving-syntax.md:9]: I think we should expand a bit on migration

**Review by nikomatsakis — CHANGES_REQUESTED** [review]

**Comment by rylev** [comment]
@joshtriplett do you need help getting this merged? We'll soon be doing a sweep over all the sections for consistency, and I'd like this to be merged before we do that.

File: src/rust-2021/reserving-syntax.md (hunk 1 above)
Comment (joshtriplett) [reply src/rust-2021/reserving-syntax.md]: I've changed the line in the summary to make it clear that this is an identifier. The subsequent non-summary explanation goes into more detail on that point, so I think this is now addressed.

**Comment by joshtriplett** [comment]
I've updated this to address feedback. I think it's ready to merge now.

**Review by rylev — APPROVED** [review]
Looks great! We still need a section on migration, but we can do that in a separate PR.

**Comment by joshtriplett** [comment]
On Thu, Jul 22, 2021 at 01:02:34AM -0700, Ryan Levick wrote:
> Looks great! We still need a section on migration, but we can do that in a separate PR. 

Can you clarify what you're looking for on migration? I tried to cover
what the compiler's migrations would do, and I think other sections of
the edition guide cover "how to invoke migrations".

**Pull request merged**